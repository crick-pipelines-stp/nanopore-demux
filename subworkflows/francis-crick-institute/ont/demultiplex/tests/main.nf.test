nextflow_process {

    name "Test Subworkflow ONT_DEMULTIPLEX"
    script "../main.nf"
    process "ONT_DEMULTIPLEX"

    tag "subworkflows"
    tag "ont"
    tag "ont_demux"

    test("simplex_default") {
        tag "ont_demux/simplex_default"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = false
                input[3] = null
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100'
                input[7] = null
                input[8] = null
                input[9] = false
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 1 },
                { assert process.out.demux_bam.size() == 0 },
            )
        }
    }

    test("simplex_emitbam") {
        tag "ont_demux/simplex_emitbam"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = false
                input[3] = null
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100'
                input[7] = null
                input[8] = null
                input[9] = true
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 0 },
                { assert process.out.demux_bam.size() == 1 },
            )
        }
    }

    test("simple_batch") {
        tag "ont_demux/simple_batch"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = true
                input[3] = null
                input[4] = false
                input[5] = 4
                input[6] = 's3://crick-pipeline-technologies-nextflow-test-data/ont/batch_pod5'
                input[7] = null
                input[8] = null
                input[9] = false
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 1 },
                { assert process.out.demux_bam.size() == 0 },
            )
        }
    }

    test("simplex_bckit") {
        tag "ont_demux/simplex_bckit"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = "SQK-PCB114-24"
                input[2] = false
                input[3] = null
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100'
                input[7] = null
                input[8] = null
                input[9] = true
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success }
            )
        }
    }

    test("simplex_resume") {
        tag "ont_demux/simplex_resume"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = false
                input[3] = null
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100'
                input[7] = null
                input[8] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5.bam'
                input[9] = false
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 1 },
                { assert process.out.demux_bam.size() == 0 },
            )
        }
    }

    test("multiplex_default") {
        tag "ont_demux/multiplex_default"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = true
                input[3] = null
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed'
                input[7] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/1000_reads_6_samples.bam'
                input[8] = null
                input[9] = false
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/samplesheet.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 6 },
                { assert process.out.demux_bam.size() == 0 },
            )
        }
    }

    test("multiplex_parsebc") {
        tag "ont_demux/multiplex_parsebc"

        when {
            process {
                """
                input[0] = "hac"
                input[1] = null
                input[2] = true
                input[3] = 2
                input[4] = false
                input[5] = 999999
                input[6] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed'
                input[7] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/1000_reads_6_samples.bam'
                input[8] = null
                input[9] = false
                input[10] = '$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/samplesheet_parsebc.csv'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert process.out.bam.size() == 1 },
                { assert process.out.demux_fastq.size() == 6 },
                { assert process.out.demux_bam.size() == 0 },
            )
        }
    }
}

// take:
// val_model         // Specify a short code like HAC or SUP
// val_bc_kit        // The barcode kit to demultiplex against
// val_check_barcode // Specifies if the run directory should be searched for a valid barcode kit
// val_bc_parse_pos  // The parse position to substring the barcode from in the meta data
// val_append_bc     // Appends bc-kit to sample matching
// val_batch_num     // Number of files in a dorado batch
// val_run_dir       // The string path of the nanopore run directory
// val_bam           // The string path of a BAM file to use instead of basecalling
// val_resume_bam    // The string path of a BAM file to resume basecalling from
// val_emit_bam      // Defines whether demux outputs a bam or fastq file
// val_samplesheet   // The string path of the samplesheet to parse for metadata if given

// emit:
// pod5        = ch_pod5_files  // channel: [ path(pod5) ]
// bam         = ch_bam         // channel: [ val(meta), path(bam) ]
// demux_fastq = ch_demux_fastq // channel: [ val(meta), path(fastq) ]
// demux_bam   = ch_demux_bam   // channel: [ val(meta), path(bam) ]
// versions    = ch_versions    // channel: path(versions.yml)