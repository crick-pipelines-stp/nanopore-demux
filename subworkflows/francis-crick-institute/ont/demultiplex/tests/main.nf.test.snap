{
    "simplex_resume": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99",
                "versions.yml:md5,f09825a9b3f50d7f208f71cee9fbde60"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:34:36.981398"
    },
    "simplex_emitbam": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:32:02.314719"
    },
    "multiplex_parsebc": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:34:43.557866"
    },
    "multiplex_default": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:34:40.280274"
    },
    "simple_batch": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99",
                "versions.yml:md5,f09825a9b3f50d7f208f71cee9fbde60"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-09T17:00:40.184102"
    },
    "simplex_default": {
        "content": [
            [
                "versions.yml:md5,a2381d271da27bf7e7ac07e0bb733c2b",
                "versions.yml:md5,c8aa2b4d1847fd026d3af24d37133d99"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:31:39.183893"
    }
}